### ISSUE TITLE
sen2r() works on Ubuntu, but fails on Windows for reasons unknown

### ISSUE THREAD

**Issue by fdetsch** [ranghetti/sen2r#342]
I have a script that runs perfectly smooth on a Ubuntu server (`sessionInfo()` included below code):

```r
library(raster)
library(sen2r)

## area of interest
deu = getData(
  country = "DEU"
  , level = 4
  , path = tmpDir()
)

aoi = subset(
  deu
  , NAME_4 == "Wilhelmsthal"
)

## get sentinel-2 data
sen2r(
  gui = FALSE
  , timewindow = as.Date("2020-07-10")
  , extent = aoi
  , extent_name = "Wilhelmsthal"
  , list_prods = "BOA"
  , list_indices = "NDVI"
  , path_l2a = tmpDir()
  , path_out = "."
)

list.files(
  pattern = "Wilhelmsthal.*\\d{2}.tif$"
  , recursive = TRUE
)
# [1] "BOA/S2A2A_20200710_065_Wilhelmsthal_BOA_10.tif"   "NDVI/S2A2A_20200710_065_Wilhelmsthal_NDVI_10.tif"
```

<details>
  <summary>sessionInfo (Ubuntu)</summary>

  ```r
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8        LC_COLLATE=C.UTF-8    
 [5] LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8    LC_PAPER=C.UTF-8       LC_NAME=C             
 [9] LC_ADDRESS=C           LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] stars_0.4-3      abind_1.4-5      sf_0.9-5         sen2r_1.3.7.9001 raster_3.3-13    sp_1.4-2        

loaded via a namespace (and not attached):
 [1] clisymbols_1.2.0   tidyselect_1.1.0   purrr_0.3.4        lattice_0.20-41    V8_3.2.0          
 [6] vctrs_0.3.2        generics_0.0.2     XML_3.99-0.5       rlang_0.4.7        geojsonlint_0.4.0 
[11] e1071_1.7-3        pillar_1.4.6       foreign_0.8-79     httpcode_0.3.0     glue_1.4.1        
[16] DBI_1.1.0          prompt_1.0.0       foreach_1.5.0      lifecycle_0.2.0    geojsonio_0.9.2   
[21] rgeos_0.5-3        RcppTOML_0.1.6     codetools_0.2-16   doParallel_1.0.15  maptools_1.0-1    
[26] parallel_4.0.2     curl_4.3           class_7.3-17       Rcpp_1.0.5         KernSmooth_2.23-17
[31] classInt_0.4-3     jsonvalidate_1.1.0 lwgeom_0.2-5       jsonlite_1.7.0     stringi_1.4.6     
[36] dplyr_1.0.0        grid_4.0.2         jqr_1.1.0          rgdal_1.5-12       tools_4.0.2       
[41] magrittr_1.5       lazyeval_0.2.2     geojson_0.3.4      tibble_3.0.3       crul_0.9.0        
[46] crayon_1.3.4       pkgconfig_2.0.3    ellipsis_0.3.1     xml2_1.3.2         data.table_1.13.0 
[51] httr_1.4.2         rstudioapi_0.11    iterators_1.0.12   R6_2.4.1           units_0.6-7       
[56] compiler_4.0.2
  ```
</details>
 
Running the exact same code on a Windows machine succeeds in downloading the requested SAFE file, but then fails to create BOA and NDVI GeoTIFF mosaics with a more or less blank processing report (again, `sessionInfo()` below code):

```r
[2020-07-30 08:12:21] #### Starting sen2r execution. ####
[2020-07-30 08:12:22] Searching for available SAFE products on SciHub...
[2020-07-30 08:12:24] Computing output names...
Linking to GEOS 3.8.0, GDAL 3.0.4, PROJ 6.3.1
[2020-07-30 08:12:25] Starting to download the required level-2A SAFE products.
Images S2A_MSIL2A_20200710T102031_N0214_R065_T32UPA_20200710T125814.SAFE are already
on your system and will be skipped. Set "overwrite_safe" to TRUE to re-download them.
No L2A images are needed.
[2020-07-30 08:12:25] Download of level-2A SAFE products terminated.
[2020-07-30 08:12:25] Starting to download the required level-1C SAFE products.
No L1C images are needed.
[2020-07-30 08:12:25] Download of level-1C SAFE products terminated.
[2020-07-30 08:12:25] Updating output names...
[2020-07-30 08:12:25] Starting to translate SAFE products in custom format.
Using UTM zone 32N.
1 output files were correctly created.
+-------------------------------------------------------------------------------------
¦ sen2r Processing Report
¦-------------------------------------------------------------------------------------
¦ Dates to be processed based on processing parameters: 1
¦ Processing completed for: 0 out of 1 expected dates.
¦ WARNING: Outputs for: 1 out of 1 expected dates not created because of unexpected
¦ reasons."
¦ These files will be skipped during next executions from the current JSON parameter
¦ file. To try again to build them, remove their file names in the text file
¦ ".\.ignorelist.txt".
+-------------------------------------------------------------------------------------
[2020-07-30 08:12:26] #### sen2r session terminated. ####
```

<details>
  <summary>sessionInfo (Windows)</summary>

  ```r
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sf_0.9-5         raster_3.3-13    sp_1.4-2         sen2r_1.3.7.9001

loaded via a namespace (and not attached):
 [1] clisymbols_1.2.0   tidyselect_1.1.0   purrr_0.3.4        lattice_0.20-41   
 [5] V8_3.2.0           vctrs_0.3.2        generics_0.0.2     stars_0.4-3       
 [9] XML_3.99-0.5       rlang_0.4.7        geojsonlint_0.4.0  e1071_1.7-3       
[13] pillar_1.4.6       foreign_0.8-80     httpcode_0.3.0     glue_1.4.1        
[17] DBI_1.1.0          prompt_1.0.0       foreach_1.5.0      lifecycle_0.2.0   
[21] geojsonio_0.9.2    rgeos_0.5-3        RcppTOML_0.1.6     codetools_0.2-16  
[25] doParallel_1.0.15  maptools_1.0-1     parallel_4.0.2     curl_4.3          
[29] class_7.3-17       Rcpp_1.0.5         KernSmooth_2.23-17 classInt_0.4-3    
[33] jsonvalidate_1.1.0 lwgeom_0.2-5       jsonlite_1.7.0     abind_1.4-5       
[37] stringi_1.4.6      dplyr_1.0.0        grid_4.0.2         jqr_1.1.0         
[41] rgdal_1.5-12       tools_4.0.2        magrittr_1.5       lazyeval_0.2.2    
[45] geojson_0.3.4      tibble_3.0.3       crul_0.9.0         crayon_1.3.4      
[49] pkgconfig_2.0.3    ellipsis_0.3.1     xml2_1.3.2         data.table_1.13.0 
[53] httr_1.4.2         rstudioapi_0.11    iterators_1.0.12   R6_2.4.1          
[57] units_0.6-7        compiler_4.0.2
  ```
</details>

To be honest, I don't have the slightest idea what could be going wrong on Windows. Has this behavior been reported before?

**Comment by ranghetti** [comment]
Unfortunately I am not able to reproduce the error (the two Tiffs are correctly produced on a Windows machine, in my case).

Can you try providing me the missing information (#186 points 3 and 5)?

**Comment by fdetsch** [comment]
Sorry, I apparently overlooked the README. I updated the console output above and attached the result of `sen2r::load_binpaths()` below (`~` is the path to my Windows user's Documents folder):

```r
$aria2c
[1] "~\\.sen2r\\aria2c.exe"

$sen2cor
[1] "~\\.sen2r\\sen2cor\\Sen2Cor-02.05.05-win64\\L2A_Process.bat"

attr(,"path")
[1] "~\\.sen2r/paths.json"
```

**Comment by ranghetti** [comment]
The error occurs in `s2_translate()` (internally called by `sen2r()`); but I cannot investigate where and why without debugging the function, since I am not able to replicate it. If you have debugging experience you can try by yourself.

**Comment by fdetsch** [comment]
Turns out that the full paths of my Level-2A `.SAFE.zip` contents were too long for [`unzip()`](https://github.com/ranghetti/sen2r/blob/devel/R/s2_download.R#L293) to handle. When I first ran the code on Windows, `sen2r()` must have encountered the following error (which I obviously ignored):

```r
[2020-07-30 16:28:44] #### Starting sen2r execution. ####
Loading required namespace: stringi
Registered S3 method overwritten by 'geojsonlint':
  method         from 
  print.location dplyr
[2020-07-30 16:28:44] Searching for available SAFE products on SciHub...
[2020-07-30 16:28:47] Computing output names...
Linking to GEOS 3.8.0, GDAL 3.0.4, PROJ 6.3.1
[2020-07-30 16:28:48] Starting to download the required level-2A SAFE products.
[2020-07-30 16:28:48] Check if products are available for download...
[2020-07-30 16:28:48] 1 Sentinel-2 images are already online.
[2020-07-30 16:28:48] Downloading Sentinel-2 image 1 of 1
        (S2A_MSIL2A_20200710T102031_N0214_R065_T32UPA_20200710T125814.SAFE)...
  |=======================================================================================| 100%
Error in { : 
  task 1 failed - "task 1 failed - "task 1 failed - "cannot open file '../2020_bark_beetle_monitoring/00_basedata/001_data/sentinel-2/safe/level-2a/S2A_MSIL2A_20200710T102031_N0214_R065_T32UPA_20200710T125814.SAFE/GRANULE/L2A_T32UPA_A026374_20200710T102027/QI_DATA/T32UPA_20200710T102031_PVI.jp2': No such file or directory"""
```

At this point, everything until file `GRANULE/L2A_T32UPA_A026374_20200710T102027/QI_DATA/T32UPA_20200710T102031_PVI.jp2` had been extracted into a folder `S2A_MSIL2A_20200710T102031_N0214_R065_T32UPA_20200710T125814.SAFE`. The next time I ran `sen2r()`, the function assumed that all data was already there (since the folder existed), and hence skipped the file altogether (ie. didn't throw an error). Ultimately, this broke `s2_translate()` which couldn't then find all files it requires.

Long story short, I moved my project to a shorter Windows folder path and the code now completes without any issue. I only wish that the whole "No such file or directory" `unzip()` error would have been a bit less cryptic in the first place..

**Comment by ranghetti** [comment]
Thank you for the detailed report.
I will have a look to this case in future, while `s2_translate()` actually would not need the PVI jp2 file (the validity of the SAFE archive is checked before using it by checking the required jp2 files - band images - to be present). Unfortunately, it is not hard to exceed the maximum allowed path length in Windows, considering how long the SAFE names (and subdirectory names) are...
